### PR TITLE
Expand documentation for Cartesian projections

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to expand the documentations of the Cartesian [linear](https://www.pygmt.org/dev/projections/nongeo/cartesian_linear.html), [logarithmic](https://www.pygmt.org/dev/projections/nongeo/cartesian_logarithmic.html), and [power](https://www.pygmt.org/dev/projections/nongeo/cartesian_power.html) projections regarding the scale option.

Currently, the general syntax is repeated for the scale option (lower-case letter version), as we have it for nearly all other projections. Or do we only want to state the width-height version (upper-case letter version), as is currently done for the [polar projection](https://www.pygmt.org/dev/projections/nongeo/polar.html)?

**For context / first mentioned**: https://github.com/GenericMappingTools/pygmt/pull/2641#discussion_r1305791567

**Preview**:
- Cartesian [linear](https://pygmt-dev--2643.org.readthedocs.build/en/2643/projections/nongeo/cartesian_linear.html)
- Cartesian [logarithmic](https://pygmt-dev--2643.org.readthedocs.build/en/2643/projections/nongeo/cartesian_logarithmic.html)
- Cartesian [power](https://pygmt-dev--2643.org.readthedocs.build/en/2643/projections/nongeo/cartesian_power.html)

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: examples/projections/nongeo/cartesian_linear.py [hunk 1]
```diff
@@ -2,8 +2,11 @@
 Cartesian linear
 ================
 
-**X**\ *width*/[*height*]: Give the *width* of the figure and the optional
-*height*.
+**X**\ *width*\ [/*height*] or **x**\ *x-scale*\ [/*y-scale*]
+
+Give the *width* of the figure and the optional *height*.
+The lower-case version **x** is similar to **X** but expects
+a *x-scale* and an optional *y-scale*.
```
Comment (michaelgrund) [inline examples/projections/nongeo/cartesian_linear.py:9]:
```suggestion
an *x-scale* and an optional *y-scale*.
```

File: examples/projections/nongeo/cartesian_logarithmic.py [hunk 2]
```diff
@@ -2,9 +2,14 @@
...
 
-**X**\ *width*\ [**l**]/[*height*\ [**l**]]: Give the *width* of the figure and
-the optional *height*. Each axis with a logarithmic transformation
-requires **l** after its size argument.
+**X**\ *width*\ [**l**][/*height*\ [**l**]] or
+**x**\ *x-scale*\ [**l**][/*y-scale*\ [**l**]]
+
+Give the *width* of the figure and the optional *height*.
+The lower-case version **x** is similar to **X** but expects
+a *x-scale* and an optional *y-scale*.
```
Comment (michaelgrund) [inline examples/projections/nongeo/cartesian_logarithmic.py:10]:
```suggestion
an *x-scale* and an optional *y-scale*.
```

File: examples/projections/nongeo/cartesian_power.py [hunk 3]
```diff
@@ -2,10 +2,14 @@
...
-**X**\ *width*\ [**p**\ *pvalue*]/[*height*\ [**p**\ *pvalue*]]: Give the
-*width* of the figure and the optional argument *height*. Each axis with
-a power transformation requires **p** and the exponent for that axis
-after its size argument.
+**X**\ *width*\ [**p**\ *pvalue*][/*height*\ [**p**\ *pvalue*]] or
+**x**\ *x-scale*\ [**p**\ *pvalue*][/*y-scale*\ [**p**\ *pvalue*]]
+
+Give the *width* of the figure and the optional argument *height*.
+The lower-case version **x** is similar to **X** but expects
+a *x-scale* and an optional *y-scale*.
```
Comment (michaelgrund) [inline examples/projections/nongeo/cartesian_power.py:10]:
```suggestion
an *x-scale* and an optional *y-scale*.
```

**Review by michaelgrund — APPROVED** [review]

File: examples/projections/nongeo/cartesian_linear.py (hunk 1 above)
Comment (yvonnefroehlich) [reply examples/projections/nongeo/cartesian_linear.py]: These typo fixes are included in https://github.com/GenericMappingTools/pygmt/pull/2643/commits/943ad7fe6f80db3e2bdb320d0f1d46c12802a29c, in which I added some text to mention the possibility of using geographic as well as data/time input data.

**Review by seisman — APPROVED** [review]

**Pull request merged**